include/baresip.h

```diff
@@ -283,18 +283,28 @@ enum presence_status {
 	PRESENCE_BUSY
 };
 
+
+struct contact;
+typedef void (contact_update_h)(struct contact *c, bool remove, void *arg);
+
 struct contacts {
 	struct list cl;
 	struct hash *cht;
+
+	contact_update_h* handler;
+	void* handler_arg;
 };
 
-struct contact;
 
 int  contact_init(struct contacts *contacts);
 void contact_close(struct contacts *contacts);
 int  contact_add(struct contacts *contacts,
 		 struct contact **contactp, const struct pl *addr);
+void contact_remove(struct contacts* contacts, struct contact* c);
```

Comment:
please note that the pointer star `*` should be oriented to the right, just like
the rest of the source code. (multiple places).